pkg/controller/job/job_controller_test.go

```diff
@@ -370,6 +372,19 @@ func TestControllerSyncJob(t *testing.T) {
 			expectedFailed:     1,
 			expectedPodPatches: 1,
 		},
+		"no new pod; possible finalizer update of failed pod": {
+			wFinalizersExclusive: pointer.Bool(true),
```

Comment:
this line is probably not needed since you fixed the legacy path too?

Reply:
To make this work for the legacy path, we'll need to set `failedPods` to 1.

But setting `failedPods` to 1 would break the case with finalizers because of how failed pods are created - https://github.com/kubernetes/kubernetes/pull/114516#discussion_r1060931512